gdnative-core/src/export/class_builder.rs

```diff
@@ -33,202 +33,20 @@
-                            );
-                            return ptr::null_mut();
-                        }
-                    };
-
-                    let wrapper = C::UserData::new(val);
-                    C::UserData::into_user_data(wrapper) as *mut _
-                }
-
-                sys::godot_instance_create_func {
-                    create_func: Some(constructor::<C>),
-                    method_data: ptr::null_mut(),
-                    free_func: None,
-                }
-            };
-
-            let destroy = {
-                unsafe extern "C" fn destructor<C: NativeClass>(
-                    _this: *mut sys::godot_object,
-                    _method_data: *mut libc::c_void,
-                    user_data: *mut libc::c_void,
-                ) {
-                    if user_data.is_null() {
-                        godot_error!(
-                            "gdnative-core: user data pointer for {} is null (did the constructor fail?)",
-                            C::class_name(),
-                        );
-                        return;
-                    }
-
-                    let wrapper = C::UserData::consume_user_data_unchecked(user_data);
-                    drop(wrapper)
-                }
-
-                sys::godot_instance_destroy_func {
-                    destroy_func: Some(destructor::<C>),
-                    method_data: ptr::null_mut(),
-                    free_func: None,
-                }
-            };
-
-            if is_tool {
-                (get_api().godot_nativescript_register_tool_class)(
-                    self.handle as *mut _,
-                    class_name.as_ptr() as *const _,
-                    base_name.as_ptr() as *const _,
-                    create,
-                    destroy,
-                );
-            } else {
-                (get_api().godot_nativescript_register_class)(
-                    self.handle as *mut _,
-                    class_name.as_ptr() as *const _,
-                    base_name.as_ptr() as *const _,
-                    create,
-                    destroy,
-                );
-            }
-
-            (get_api().godot_nativescript_set_type_tag)(
-                self.handle as *mut _,
-                class_name.as_ptr() as *const _,
-                crate::nativescript::type_tag::create::<C>(),
-            );
-
-            let builder = ClassBuilder {
-                init_handle: self.handle,
-                class_name,
-                _marker: PhantomData,
-            };
-
-            C::register_properties(&builder);
-
-            // register methods
-            C::register(&builder);
-        }
-    }
-}
-
 #[derive(Debug)]
 pub struct ClassBuilder<C> {
-    init_handle: *mut libc::c_void,
-    class_name: CString,
-    _marker: PhantomData<C>,
+    pub(crate) init_handle: *mut libc::c_void,
+    pub(crate) class_name: CString,
+    pub(crate) _marker: PhantomData<C>,
 }
```

Comment:
Perhaps add a constructor so the `pub(crate)`s aren't necessary? IIRC it isn't necessary to access those fields from outside except for constructing the type.

Reply:
Actually, `PropertyBuilder::done()` is now in a different file than `ClassBuilder` as well, and it needs access to the fields:
```rs
unsafe {
    (get_api().godot_nativescript_register_property)(
        self.class_builder.init_handle,
        self.class_builder.class_name.as_ptr(),
        path.as_ptr() as *const _,
        &mut attr,
        set,
        get,
    );
}
```

Reply:
Interesting. It feels quite awkward this way, but I can't say I have a better idea. Perhaps the contents of `class_builder` can be moved into the `export` module itself, given how short that module currently is, or maybe this is calling for some new abstraction like a `RawClassBuilder` type that can be more easily shared between sub-modules of `export`, without exposing its internals to the entire crate.

It can also be said that making the `Export` trait the only top-level item in the `export` module might give new contributors a wrong idea about its purpose, but that isn't a urgent matter here, since it's a purely internal problem (code organization, vs. item organization).

Reply:
Solved it by making the fields `pub(super)`. Still added a constructor.

Reply:
Ah wait, your comment only showed now.

> It can also be said that making the `Export` trait the only top-level item in the `export` module might give new contributors a wrong idea about its purpose, but that isn't a urgent matter here, since it's a purely internal problem (code organization, vs. item organization).

True. I also did it because Clippy has something against nested modules with the same name as their parents, and `export::export` would have been the name standing to reason. `class_builder.rs` would be the alternative, but as you say, we can do the internal organization also later, without affecting the API.

---

For reference, the public API of `gdnative::export` now looks as follows (I should really find time to do the generate-doc-for-each-pull-request CI at some point):
![grafik](https://user-images.githubusercontent.com/708488/140744409-185f581a-fa27-42d7-a3b7-caa741b0c515.png)

Reply:
`property` could be the place, since the `Export` trait is really only about exporting properties to the editor. I don't think it's used anywhere else -- signals in the future (#813), perhaps, but not right now.

Reply:
Moved it to `properties.rs` in this PR; otherwise it likely won't be done or there's a lot of bureaucracy for a little change.
Thanks for the feedback!